src/document_to_podcast/preprocessing/data_loaders.py

```diff
@@ -36,6 +38,34 @@ def load_docx(docx_file: str | UploadedFile) -> str | None:
         return None
 
 
+def load_file(file: str | UploadedFile) -> str | None:
+    """
+    Loads the content of a file or URL and converts it to markdown.
+
+    Args:
+        file (str | UploadedFile): The path to the file, a URL.
+
+    Returns:
+        str | None: The Markdown text content, or None if an error occurs.
+    """
+    try:
+        markdown_converter = MarkItDown()
+        if isinstance(file, str):  # for URL and filepath
+            if not os.path.exists(file):
+                logger.error(f"File not found: {file}")
+                return None
```

Comment:
For `url` we have the `load_url` function so this isnt really applicable here I think.

Reply:
While I was working on this PR, I didn't know if someone else already added that functionality. Hence I added URL as a source as well to make it more streamlined.

Reply:
Removed the url part.